src/nbl/ext/RadixSort/RadixSort.cpp

```diff
@@ -0,0 +1,76 @@
+#include "nbl/ext/RadixSort/RadixSort.h"
+
+namespace nbl
+{
+namespace ext
+{
+namespace RadixSort
+{
+
+RadixSort::RadixSort(video::IDriver* driver, const uint32_t wg_size) : m_wg_size(wg_size)
+{
+	assert(nbl::core::isPoT(m_wg_size));
+
+	const asset::SPushConstantRange pc_range = { asset::ISpecializedShader::ESS_COMPUTE, 0u, sizeof(nbl_glsl_ext_RadixSort_Parameters_t) };
+
+	{
+		const uint32_t count = 2u;
+		video::IGPUDescriptorSetLayout::SBinding binding[count];
+		for (uint32_t i = 0; i < count; ++i)
+			binding[i] = { i, asset::EDT_STORAGE_BUFFER, 1u, video::IGPUSpecializedShader::ESS_COMPUTE, nullptr };
+
+		m_histogram_ds_layout = driver->createGPUDescriptorSetLayout(binding, binding + count);
+		m_histogram_pipeline_layout = driver->createGPUPipelineLayout(&pc_range, &pc_range + 1, core::smart_refctd_ptr(m_histogram_ds_layout));
+
+		m_histogram_pipeline = driver->createGPUComputePipeline(nullptr, core::smart_refctd_ptr(m_histogram_pipeline_layout),
+			createShader("nbl/builtin/glsl/ext/RadixSort/default_histogram.comp", driver));
+	}
+
+	{
+		const uint32_t count = 3u;
+		video::IGPUDescriptorSetLayout::SBinding binding[count];
+		for (uint32_t i = 0; i < count; ++i)
+			binding[i] = { i, asset::EDT_STORAGE_BUFFER, 1u, video::IGPUSpecializedShader::ESS_COMPUTE, nullptr };
+
+		m_scatter_ds_layout = driver->createGPUDescriptorSetLayout(binding, binding + count);
+		m_scatter_pipeline_layout = driver->createGPUPipelineLayout(&pc_range, &pc_range + 1, core::smart_refctd_ptr(m_scatter_ds_layout));
+
+		m_scatter_pipeline = driver->createGPUComputePipeline(nullptr, core::smart_refctd_ptr(m_scatter_pipeline_layout),
+			createShader("nbl/builtin/glsl/ext/RadixSort/default_scatter.comp", driver));
+	}
+}
+
+core::smart_refctd_ptr<video::IGPUSpecializedShader> RadixSort::createShader(const char* shader_file_path, video::IDriver* driver)
+{
+	const char* source_fmt =
+R"===(#version 450
+
+#define _NBL_GLSL_WORKGROUP_SIZE_ %u
+#define BITS_PER_PASS 4
+#define NUM_BUCKETS 16
```

Comment:
can you please suck these up from the constexprs you've defined in `RadixSort.h` so that I we can try it with 4,5 and 6 bits?